src/components/ColorPicker/components/AlphaField/AlphaField.tsx

```diff
@@ -0,0 +1,63 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import {clamp} from '@shopify/javascript-utilities/math';
+
+import {useI18n} from '../../../../utilities/i18n';
+import {TextField} from '../../../TextField';
+import styles from '../../ColorPicker.scss';
+
+export interface AlphaFieldProps {
+  alpha: number;
+  onChange(alpha: number): void;
+}
+
+export function AlphaField({alpha, onChange}: AlphaFieldProps) {
+  const i18n = useI18n();
+
+  const initialValue = Math.round(alpha * 100);
+  const [percentage, setPercentage] = useState(
```

Comment:
We can use lazy initial state here

```tsx
  const [percentage, setPercentage] = useState(() => {
    const initialValue = Math.round(alpha * 100);
    return initialValue === null ? 0 : initialValue,
  })
```

Or a slightly similar approach using the conditional or operator

```tsx
  const [percentage, setPercentage] = useState(() => Math.round(alpha * 100); || 0)
```

However since math.round isn't very expensive we could just it down too

```tsx
  const [percentage, setPercentage] = useState(Math.round(alpha * 100); || 0)
```

Reply:
Nice!